# Ejercicio 1

contar el número de líneas de los ficheros bookings.csv.bz2 y searches.csv.bz2

In [1]:
import bz2
import pandas as pd
import numpy as np

In [7]:
bookings_path = "/home/dsc/data/challenge/bookings.csv.bz2"
searches_path = "/home/dsc/data/challenge/searches.csv.bz2"

### Enfoque 1
Usando directamente python la librería bz2 permite leer el fichero sin descomprimirlo antes.
Se piden líneas y se incrementa el contador.

TODO: Cambiar con with para no tener que gestionar explícitamente el cierre tras error

In [18]:
bookings = bz2.open(bookings_path, "r")
count = 0
for line in bookings:
    count += 1
bookings.close()
count

10000011

In [19]:
searches = bz2.open(searches_path, "r")
count = 0
for line in searches:
    count += 1
searches.close()
count

20390199

### Enfoque 2

Usando comandos de shell y pipes. El {} sirve para pasar variables de python al shell

In [8]:
!bunzip2 -c {bookings_path} | wc -l

10000011


In [9]:
!bunzip2 -c {searches_path} | wc -l

20390198


#### Nota: No sale exactamente lo mismo!
| Metodo | resultado |
|---|---|
| Python | 20,390,199 | 
| Shell  | 20,390,198 |  
